src/client/transports/json_rpc_transport.ts

```diff
@@ -354,9 +359,13 @@ export class JsonRpcTransport implements A2ATransport {
         jsonData,
         e
       );
-      throw new Error(
-        `Failed to parse SSE event data: "${jsonData.substring(0, 100)}...". Original error: ${e.message}`
-      );
+      if (e instanceof SyntaxError) {
+        throw new Error(
+          `Failed to parse SSE event data: "${jsonData.substring(0, 100)}...". Original error: ${e.message}`
+        );
+      } else {
+        throw e;
+      }
```

Comment:
Feels like a behavior change, previously all errors were wrapped here. I think this can be refactored to avoid `startsWith` matching above similar to #167 which should also make this block cleaner. Can we avoid checking for `SyntaxError` specifically for now and refactor it in a dedicated PR?